packages/engine/errors/7054.md

```diff
@@ -0,0 +1,34 @@
+---
+original: "No index signature with a parameter of type '{0}' was found on type '{1}'."
+excerpt: "You are trying to access an object using a key with type 'A,' but I am looking for a property's value type on type 'B.'"
```

Comment:
I'm not sure this is very accurate 🤔 
First, let's address the typos haha

- `'A,'` contains extra comma `,` before end of single quote
- `'B,'` contains extra colon `.` before end of single quote
- The second part of the sentence is wrong. It has nothing to do with the property's value but with the object's index signature (keyof)

The original diagnostic has two key parts:
- `(...) parameter of type '{0}'` - implies it's a function and it also implies you're using one of this function's parameters as an index to access a property. It'd be good to mention it somehow in the description.
- `No index signature (...) was found on type '{1}'.` - it more or less (inaccurately) says "object type `{1}` doesn't have a key that matches `{0}` type"

So I'd suggest to go with something along these lines:
```
excerpt: You cannot use type '{0}' to access properties of `{1}`
```

As far as I know `{0}` can only be `string` or `number` or `symbol` in this diagnostic. Literal types show different diagnostic, and types that cannot be used as index also error with a difference diagnostic, so it should be OK to preserve the values of `{0}` and `{1}` in the excerpt. Otherwise, made up types won't be particularly useful here.